### DISCUSSION TITLE
WeibullAFTFitter.predict_cumulative_hazard

### DISCUSSION THREAD

**Discussion by rsmita** [CamDavidsonPilon/lifelines#1445] [Q&A]
Hi,

I understand that **predict_cumulative_hazard**() function is to get the cumulative probability of failure. However I am unable to interpret the output from this function. I expected that highest probability would be 100%, howver I am getting highest failure probability as 2.35. 

Can you please help interpreting the below output from predict_cumulative_hazard(). 

_# Code
aft = WeibullAFTFitter()
aft.fit(df_refined, duration_col='Age', event_col='Failed', ancillary=df_refined)
X = df_refined.loc[0]
aft.predict_cumulative_hazard(X, ancillary=X)_

#Output
	0
6.358904	0.014141
6.917808	0.017875
7.939726	0.026225
8.345205	0.030122
10.000000	0.049824
...	...
37.000000	1.897246
38.000000	2.043352
39.000000	2.196473
39.200000	2.227952
**40.000000**	**2.356753**

Thank you,

**Comment by CamDavidsonPilon** [comment]
Cumulative hazards aren't probabilities. You can convert to probabilities by taking the negative exponential, eg: `exp(-cum_haz(t))` is a probabilities, specifically the survival function, `S(t)`

**Reply by rsmita** [reply]
Hi, appreciate your response. 

I guess cumulative Hazard function will not help to calculate probability of failure of an individual item at time,t. 
I specifically need to predict:
1. Probability of machine failing at age t
2. How many machines will fail in next month, considering their current age.

Here for 1st one, can we use predict_survival_function() to get S(t) & calculate F(t) = 1-S(t) ?
Which function can help answer the 2nd point?

Thank you,

**Reply by CamDavidsonPilon** [reply]
1. F(t) will tell you has it not failed _yet_, not "failing at age t". "failing at age t" would be something like S(t) - S(t-1)


2. Try using something like https://lifelines.readthedocs.io/en/latest/Survival%20Regression.html#prediction-on-censored-subjects

**Comment by rsmita** [comment]
Thanks you, Had applied it on my data & got the survival prediction
probability, but unable to interpret & answer to following:
1. What is the probability of failure at current age+ 6 months (

Would you be able to guide me on this part please.

Thanks & Regards,
Smita

**Comment by rsmita** [comment]
Hi,

Thank you for your response.

In that case, can we calculate probability of failure as 1-probability of
survival i.e. 1-S(t)

Thanks,
Smita

**Reply by CamDavidsonPilon** [reply]
Yes, that's correct
